controller/soap_web_service.rst

```diff
@@ -50,56 +50,68 @@ In this case, the SOAP service will allow the client to call a method called
         }
     }
 
-Next, you can train Symfony to be able to create an instance of this class.
-Since the class sends an email, it's been designed to accept a ``Swift_Mailer``
-instance. Using the Service Container, you can configure Symfony to construct
-a ``HelloService`` object properly:
+Next, make sure that your new class is registered as a service. If you use
+:doc:`autowiring </service_container/autowiring>` (enabled by default), this is easy:
 
 .. configuration-block::
 
     .. code-block:: yaml
 
         # app/config/services.yml
         services:
-            hello_service:
-                class: Acme\SoapBundle\Services\HelloService
-                arguments: ['@mailer']
+            _defaults:
+                # be sure autowire is enabled
+                autowire: true
+
+            # load services from the Service directory
+            AppBundle\:
+                resource: '../../src/AppBundle/{Service,Updates,Command,Form,EventSubscriber,Twig,Security}'
 
     .. code-block:: xml
 
         <!-- app/config/services.xml -->
-        <services>
-            <service id="hello_service" class="Acme\SoapBundle\Services\HelloService">
-                <argument type="service" id="mailer"/>
-            </service>
-        </services>
+        <?xml version="1.0" encoding="UTF-8" ?>
+        <container xmlns="http://symfony.com/schema/dic/services"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services
+                http://symfony.com/schema/dic/services/services-1.0.xsd">
+
+            <services>
+                <defaults autowire="true" autoconfigure="true" />
+
+                <!-- load services from the Service directory -->
+                <prototype namespace="AppBundle\" resource="../../src/AppBundle/{Service,Updates,Command,Form,EventSubscriber,Twig,Security}" />
+            </services>
+        </container>
 
     .. code-block:: php
 
         // app/config/services.php
-        use Acme\SoapBundle\Services\HelloService;
+        use AppBundle\Service\HelloService;
 
-        $container
-            ->register('hello_service', HelloService::class)
-            ->addArgument(new Reference('mailer'));
+        $container->autowire(HelloService::class)
+            ->setPublic(false);
 
 Below is an example of a controller that is capable of handling a SOAP
-request. If ``indexAction()`` is accessible via the route ``/soap``, then the
-WSDL document can be retrieved via ``/soap?wsdl``.
-
-.. code-block:: php
+request. BEcause ``indexAction()`` is accessible via ``/soap``, the WSDL document
+can be retrieved via ``/soap?wsdl``::
 
-    namespace Acme\SoapBundle\Controller;
+    namespace AppBundle\Controller;
 
     use Symfony\Bundle\FrameworkBundle\Controller\Controller;
     use Symfony\Component\HttpFoundation\Response;
+    use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
+    use AppBundle\Service\HelloService;
```

Comment:
Ah, they *do* follow that standard in core. I suppose we could do, but I'll worry about that later ;)